src/pages/CountrySelectorPage.js

```diff
@@ -0,0 +1,86 @@
+import _ from 'underscore';
+import lodashGet from 'lodash/get';
+import React, {useCallback, useMemo} from 'react';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import * as PersonalDetails from '../libs/actions/PersonalDetails';
+import ROUTES from '../ROUTES';
+import withLocalize, {withLocalizePropTypes} from '../components/withLocalize';
+import Navigation from '../libs/Navigation/Navigation';
+import compose from '../libs/compose';
+import ONYXKEYS from '../ONYXKEYS';
+import OptionsSelectorWithSearch, {greenCheckmark} from '../components/OptionsSelectorWithSearch';
+
+const propTypes = {
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            backTo: PropTypes.string,
+        }),
+    }).isRequired,
+
+    /** User's private personal details */
+    privatePersonalDetails: PropTypes.shape({
+        /** User's home address */
+        address: PropTypes.shape({
+            country: PropTypes.string,
+        }),
+    }),
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    privatePersonalDetails: {
+        address: {
+            state: '',
+        },
+    },
+};
+
+function CountrySelectorPage(props) {
+    const translate = props.translate;
+    const route = props.route;
+    const currentCountry = route.params.countryISO || lodashGet(props.privatePersonalDetails, 'address.country');
+    const selectedSearchCountry = PersonalDetails.getCountryNameBy(currentCountry);
+
+    const countries = useMemo(
+        () =>
+            _.map(translate('allCountries'), (countryName, countryISO) => ({
+                value: countryISO,
+                keyForList: countryISO,
+                text: countryName,
+                customIcon: currentCountry === countryISO ? greenCheckmark : undefined,
+            })),
+        [translate, currentCountry],
+    );
+
+    const updateCountry = useCallback((selectedCountry) => {
+        Navigation.navigate(`${ROUTES.SETTINGS_PERSONAL_DETAILS_ADDRESS}?countryISO=${selectedCountry.value}`, 'NAVIGATE');
+    }, []);
```

Comment:
Move this function outside or just inline it. I don't think `useCallback` is doing much good.

Reply:
Resolved